CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua

```diff
@@ -0,0 +1,191 @@
+	masterFrame.Position = UDim2.new(0, 0, 0, 0)
+	masterFrame.Size = UDim2.new(1, 0, 1, 0)
+  masterFrame.Selectable = false
+  masterFrame.BackgroundTransparency = 0.8
+  masterFrame.Active = false  
+  masterFrame.ZIndex = 0
+  
+  -- FIXME(dbanks)
+  -- Debug, can see the whole frame.
+	-- masterFrame.BackgroundColor3 = Color3.new(0, 0.5, 0.5)
+	-- masterFrame.BackgroundTransparency = 0.8  
+end
+
+function ConfigureStatButtonsInMasterFrame()
+  -- Set up the row of buttons across the top and handler for button press.
+  for i, displayType in ipairs(StatsUtils.AllStatDisplayTypes) do
+    AddButton(displayType, i)
+  end
+end
+
+function OnButtonToggled(toggledDisplayType) 
+  local toggledButton = statsButtonsByType[toggledDisplayType]
+  local selectedState = toggledButton._isSelected
+  selectedState = not selectedState
+  
+  if (selectedState) then 
+    currentDisplayType = toggledDisplayType
+  else
+    currentDisplayType = nil
+  end
+  
+  UpdateButtonSelectedStates()
+  UpdateViewerVisibility()
+end
+
+function UpdateButtonSelectedStates()
+  for i, buttonType in ipairs(StatsUtils.AllStatDisplayTypes) do
+      local button = statsButtonsByType[buttonType]
+      button:SetIsSelected(buttonType == currentDisplayType)
+  end  
+end
+
+function UpdateViewerVisibility()
+  -- If someone is on, show the Viewer.
+  -- FIXME(dbanks)
+  -- Configure with details of the dude currently selected.  
+  if (currentDisplayType == nil) then 
+    statsViewer:SetVisible(false)
+    statsViewer:SetStatsAggregator(nil)
+  else
+    statsViewer:SetStatsDisplayType(currentDisplayType)
+    local aggregatorType = StatsUtils.DisplayTypeToAggregatorType[currentDisplayType]
+    statsViewer:SetStatsAggregator(allStatsAggregators:GetAggregator(aggregatorType))
+    
+    statsViewer:SetVisible(true)
+  end
+end
+
+function AddButton(displayType, index) 
+  -- Configure size and position of button.
+  -- Configure callback behavior to toggle
+  --    button on or off and show/hide viewer.
+  -- Parent button in main screen.
+  local button = statsButtonsByType[displayType]
+  
+  button:SetParent(masterFrame)
+  local aggregatorType = StatsUtils.DisplayTypeToAggregatorType[displayType]
+  button:SetStatsAggregator(
+    allStatsAggregators:GetAggregator(aggregatorType))
+  
+  local fraction = 1.0/StatsUtils.NumButtonTypes
+  local size = UDim2.new(fraction, 0, 0, StatsUtils.ButtonHeight)
+  local position = UDim2.new(fraction * (index - 1), 0, 0, 0)
+  button:SetSizeAndPosition(size, position)
+  
+  button:SetToggleCallbackFunction(OnButtonToggled)
+end
+
+function ConfigureStatViewerInMasterFrame()
+  -- Set up the widget that shows currently selected button.
+  statsViewer:SetParent(masterFrame)
+  
+  local size = UDim2.new(0.5, 0, 0.5, 0)
+  local position = UDim2.new(0.5, 0, 0.25, 0)
+  statsViewer:SetSizeAndPosition(size, position)
+end
+
+function UpdatePerformanceStatsVisibility() 
+  local localPlayer = PlayersService.LocalPlayer
```

Comment:
We create a localPlayer variable at the top of this script. Do we need that variable if we are just fetching it here